### ISSUE TITLE
intdir / outdir folders are created in the wrong folder

### ISSUE THREAD

**Issue by slhultgren** [Open-CMSIS-Pack/devtools#54]
If --intdir=... / --outdir=... are not specified on the command line to cbuildgen, these folders will be created in the parent folder instead.

The reason for this is because the m_projectDir does not contain a trailing slash, which means that when concatenating it with a path, the result will be missing a separator for the last segment.

Steps to reproduce:
* Create a test.cprj that specifies 
```xml
<output intdir="int" outdir="out" ... />
```
* Run `.../test$ cbuildgen test.cprj cmake`
* Observe that `../testint` and `../testout` was created instead of `../test/int` and `../test/out`

**Comment by brondani** [comment]
Hi @slhultgren, thanks for reporting this!

The issue seems to be a side effect of the `std::experimental::filesystem::path::remove_filename` function that removes the last directory separator, while the `std::filesystem::path::remove_filename` preserves it.
Can you please confirm that you are building with `std::experimental::filesystem`?
`std::experimental::filesystem` is deprecated is will be removed by most compiler vendors.

If we still need to support the experimental version an alternative to your suggestion could be to replace string concatenations by `std::filesystem::path::append` [operators](https://en.cppreference.com/w/cpp/filesystem/path/append) that should properly handle the directory separator.
What do you think?

**Comment by slhultgren** [comment]
@brondani thanks, good catch, it does indeed seem to be an issue with the compiler itself.

I was originally building with "-G "Visual Studio 15 2017"", which resulted in "build engine 15.9" and this shows the issue.

I upgraded to "-G "Visual Studio 16 2019"" which has "build engine 16.11" and seems to solve the issue without the PR, so I closed my PR as invalid.

Maybe the README.md should update to only have "Visual Studio 2019"?
